tvm/src/codegen/llvm/llvm_module.cc

```diff
@@ -150,7 +150,29 @@ class LLVMModuleNode final : public runtime::ModuleNode {
                            llvm::MDString::get(*ctx_, target));
     target_ = target;
     mptr_ = module_.get();
-    // this->SaveToFile("test.ll", "ll");
+    this->SaveToFile("test.ll", "ll");
+  }
+
+  void InitHB(Array<LoweredFunc> funcs, std::string target) {
+    InitializeLLVM();
+    CHECK(target.length() >= 11 && target.substr(0, 11) == "hammerblade");
+    std::ostringstream config;
+    config << "-mtriple=riscv32-unknown-unknown -mabi=ilp32f";
+    tm_ = GetLLVMTargetMachine(config.str());
+    std::unique_ptr<CodeGenLLVM> cg = CodeGenLLVM::Create(tm_);
+    ctx_ = std::make_shared<llvm::LLVMContext>();
+    std::unique_ptr<llvm::LLVMContext> ctx(new llvm::LLVMContext());
+    cg->Init(funcs[0]->name, tm_, ctx_.get(), false, false);
+    for (LoweredFunc f : funcs) {
+      cg->AddFunction(f);
+    }
+    cg->AddMainFunction(funcs[0]->name);
+    module_ = cg->Finish();
+    module_->addModuleFlag(llvm::Module::Warning, "tvm_target",
+                         llvm::MDString::get(*ctx_, target));
+    target_ = target;
+    mptr_ = module_.get();
+    this->SaveToFile("hb_test.ll", "ll");
   }
```

Comment:
This is for HB simulation in HCL right? I would suggest using another PR to introduce HB backend in HCL